src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2626,6 +2626,12 @@ def process_declarations(self, parent, nodes, arg_list,
                     root_name="_PSYCLONE_INTERNAL_NAMELIST",
                     symbol_type=DataSymbol,
                     datatype=UnsupportedFortranType(str(node)))
+
+            elif isinstance(node, Fortran2003.Data_Stmt):
+                parent.symbol_table.new_symbol(
```

Comment:
Comment please :-) c.f. Namelist_Stmt above.

Reply:
If you could include an example with an implicit loop here as justification/explanation of this approach then that would be great.

Reply:
Done (well, in the description of the new function I have added)